app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -241,25 +245,38 @@ def build_metric_id(type, resource, metric_id)
     # by ems_ref, which in Hawkular terms is the
     # fully qualified resource path from Hawkular inventory
     def run_generic_operation(operation, ems_ref, parameters = {})
-      with_provider_connection do |connection|
         the_operation = {
           :operationName => operation,
           :resourcePath  => ems_ref.to_s,
           :parameters    => parameters
         }
+        run_operation(the_operation)
+    end
 
+    def run_specific_operation(operation_name, ems_ref, parameters = {})
+      parameters[:resourcePath] = ems_ref.to_s
+      run_operation(parameters, operation_name)
+    end
+
+    def run_operation(parameters, operation_name = nil)
+      with_provider_connection do |connection|
         actual_data = {}
```

Comment:
I know, it's not your work, but is there any point of the `actual_data` variable? Same on the line https://git.io/vKAn1 It looks like a copy&pasta from the ruby gem specs.